.github/workflows/unittests.yaml

```diff
@@ -56,3 +56,52 @@ jobs:
       - name: Upload to Sonatype
         run: |
           sbt +publish
+  docs:
+    name: update tool and metric markdown docs
+    needs: test
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          # To work with the `pull_request` or any other non-`push` even with git-auto-commit
+          ref: ${{ github.head_ref }}
+      - name: Setup sdkman
+        shell: bash -l {0}
+        run: |
+          curl -s "https://get.sdkman.io" | bash
+          source "$HOME/.sdkman/bin/sdkman-init.sh"
+          sdk version
+          echo 'source "$HOME/.sdkman/bin/sdkman-init.sh"' >> "$HOME/.bash_profile"
+      - name: Install tools
+        shell: bash -l {0}
+        run: |
+          sdk install java 8.312.07.1-amzn
+          sdk install scala 2.13.0
+      - name: Build assembly JAR
+        shell: bash -l {0}
+        run: |
+          set -e
+          sbt +assembly
+      - name: Build tool docs
+        shell: bash -l {0}
+        run: |
+          set -e
+          mkdir -p docs/tools
+          java -cp target/scala-2.13/fgbio-*.jar com.fulcrumgenomics.internal.InternalTools BuildToolDocs -o docs/tools
+      - name: Build metric docs
+        shell: bash -l {0}
+        run: |
+          set -e
+          bash run_metrics_doclet.sh
+      - name: Commit generated docs
+        shell: bash -l {0}
+        run: |
+          git add docs/tools/*md
+          git add docs/*md
+          git config --global user.name "Github Actions"
+          git config --global user.email "nobody@fulcrumgenomics.com"
+          git commit --all --message "Generate docs files" --author "Nobody <nobody@fulcrumgenomics.com>" || echo "nothing to commit"
+          git push --set-upstream origin "HEAD:${{ github.head_ref }}" --follow-tags --atomic
```

Comment:
This may cause race conditions.

It will also cause quite a few merge conflicts for all developers.